### PR TITLE
Slight update to MST, and new method out of partial correlation

### PR DESCRIPTION
**Pull request by Carromattsson**
Hello,

I've added one thing to Matteo's MST code, I hope that's seamless. 

The bigger commit is a different version of partial correlation, that will be even slower if you can believe it. It's from here: https://www.ncbi.nlm.nih.gov/pubmed/21188140

This paper has a few hundred cites, so.... it may have issues, though. I couldn't test well.

Cheers,
Carolina

### CODE REVIEW

**Comment by Carromattsson** [comment]
Thanks for the testing instructions, I'll figure it out next week :)

File: netrd/reconstruction/partial_correlation_influence.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+        if index:
+            p_cor = partial_corr(TS, index=index)
+            n_TS  = p_cor.shape[0]
+            p_cor = np.delete(p_cor, index, axis=0)
+            p_cor = np.delete(p_cor, index, axis=1)
+        else:
+            p_cor = partial_corr(TS)
+
+        p_cor = np.fill_diagonal(p_cor, float("nan"))
```
Comment (sdmccabe) [inline netrd/reconstruction/partial_correlation_influence.py]: This is the problematic line. `np.fill_diagonal` modifies an array in-place and returns `None`, so the fix should be to change the line to simply `np.fill_diagonal(p_cor, float('nan'))`.

Comment (Carromattsson) [reply netrd/reconstruction/partial_correlation_influence.py]: Thanks! That's kind of inconsistent of numpy, if I'm allowed to say so. 

I think that does it? My local tests say PASSED all the way down, but I also can't really parse the warnings it's giving. Is there a sample TS object that I can just load and feed through the code line by line in a terminal window? But maybe it's whatever - it passes and we'll notice?

**Comment by Carromattsson** [comment]
I believe the change that @sdmccabe suggested fixed it?

**Comment by sdmccabe** [comment]
The warnings probably don't matter. I should clean up the tests to fix it.

**Pull request merged**